api/wasm/cpp/proxy_wasm_result.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Intrinsic enumerations available to WASM modules.
+ */
+// NOLINT(namespace-envoy)
+
+#pragma once
+
+#include <string>
+
+enum class WasmResult : uint32_t {
+  Ok = 0,
+  NotFound = 1,
+  BadType = 2,
+  BadArgument = 3,
+  SerializationFailure = 4,
+  ParseFailure = 5,
+  NoSuchStruct = 6,
+  NoSuchField = 7,
+  BadExpression = 8,
+  MemoryError = 9,
+  Eos = 10,
```

Comment:
Erm, I really don't like this being so short, but also, `end-of-stream` in proxies is mostly used in context of real stream, similar to `EOF`, but the way this is used now seems to have the meaning "no more entries in the queue", which could probably be represented by `WasmResult::Ok` + empty response.

Reply:
Empty